src/device-registry/utils/generate-filter.js

```diff
@@ -1,52 +1,130 @@
 const {
   generateDateFormat,
   generateDateFormatWithoutHrs,
-  threeMonthsBehind,
-  twoMonthsBehind,
-  oneMonthBehind,
-  threeMonthsInfront,
+  monthsBehind,
+  monthsInfront,
+  removeMonthsFromProvidedDate,
+  addMonthsToProvidedDate,
 } = require("./date");
 
 const { logObject, logElement, logText } = require("./log");
 
-const generateEventsFilter = (queryStartTime, queryEndTime, device) => {
-  if (queryStartTime && queryEndTime && !device) {
+const generateEventsFilter = (
+  queryStartTime,
+  queryEndTime,
+  device,
+  frequency
+) => {
+  let oneMonthBack = monthsBehind(1);
+  let oneMonthInfront = monthsInfront(1);
+  let defaultStartTime = generateDateFormatWithoutHrs(oneMonthBack);
+  let defaultEndTime = generateDateFormatWithoutHrs(oneMonthInfront);
+  logElement("defaultStartTime", defaultStartTime);
+  logElement("defaultEndTime", defaultEndTime);
+
+  if (queryStartTime && queryEndTime && !device && !frequency) {
+    return {
+      day: { $gte: queryStartTime, $lte: queryEndTime },
+    };
+  } else if (queryStartTime && queryEndTime && device && frequency) {
+    return {
+      day: { $gte: queryStartTime, $lte: queryEndTime },
+      "values.device": device,
+      "values.frequency": frequency,
+    };
+  } else if (!queryStartTime && !queryEndTime && device && frequency) {
+    return {
+      "values.device": device,
+      "values.frequency": frequency,
+      day: { $gte: defaultStartTime },
+    };
+  } else if (queryStartTime && !queryEndTime && !device && !frequency) {
+    return {
+      day: {
+        $gte: queryStartTime,
+        $lte: addMonthsToProvidedDate(queryStartTime, 1),
+      },
+    };
+  } else if (!queryStartTime && queryEndTime && !device && !frequency) {
+    return {
+      day: {
+        $lte: queryEndTime,
+        $gte: removeMonthsFromProvidedDate(queryEndTime, 1),
+      },
+    };
+  } else if (!queryStartTime && queryEndTime && device && frequency) {
+    return {
+      day: {
+        $lte: queryEndTime,
+        $gte: removeMonthsFromProvidedDate(queryEndTime, 1),
+      },
+      "values.device": device,
+      "values.frequency": frequency,
+    };
+  } else if (queryStartTime && !queryEndTime && device && frequency) {
+    return {
+      day: {
+        $gte: queryStartTime,
+        $lte: addMonthsToProvidedDate(queryStartTime, 1),
+      },
+      "values.device": device,
+      "values.frequency": frequency,
+    };
```

Comment:
@Baalmart, is it okay to substitute the `if else` statements with individual `if` statements?. This could improve code readability and making future changes easy.

```
let filter = {};

if(queryStartTime){
   filter['day']['$gte'] = queryStartTime
}

if(queryEndTime){
   filter['day']['$lte'] = queryEndTime
}

if(device){
   filter['values.device'] = device
}

if(frequency){
   filter['values.frequency'] = frequency
}

return filter;
```

Reply:
This makes sense @NoahNsimbe , shall make this change asap, sounds good to me.

Reply:
Done @NoahNsimbe